docs/cluster_api.md

```diff
@@ -0,0 +1,121 @@
+  - `Name [string, required]` *Name for the machine set - only used to identify the machine set within the cluster. Must be unique within the cluster*
+  - `MachineSetConfig [MachineSetConfig,inline]` *Configuration for the machine set -- see below*
+
+#### Status:
+- `AdminKubeconfig [corev1.LocalReference]` *Reference to secret that contains an admin.kubeconfig to access the cluster as cluster administrator after the cluster is up and running*
+- `Conditions [array]` *Detailed status for the cluster*
+  - `Type [string]` *Type of condition - InfraProvisioned, MasterConfigured, Running*
+  - `Status [string]` *Status of the condition - True, False, Unknown*
+  - `LastProbeTime [string]` *-optional- last time the condition was checked*
+  - `LastTransitionTime [string]` *-optional- last time the condition transitioned*
+  - `Message [string]` *-optional- Human readable message associated with the condition*
+  - `Reason [string]` *-optional- Single word camel-case reason for the condition*
+- `MachineSetCount [int]` *count of machine sets that have been created for this cluster*
+- `Provisioned [boolean]` *true if cluster-level infrastructure has been provisioned for this cluster*
+- `Running [boolean]` *true if the cluster is running and is ready to accept API requests* 
+- `MasterMachineSetName [string]` *name of the machine set designated as master*
+- `InfraMachineSetName [string]` *name of the machine set designated as infra (could be the same as master)*
+
+### MachineSet
+
+- `ObjectMetadata [metav1.ObjectMetadata]`
+- `TypeMetadata [metav1.TypeMetadata]`
+
+#### Spec
+- `MachineSetConfig [MachineSetConfig,inline]` *Configuration for the machine set (see below)*
+
+#### Status
+- `NodesProvisioned [int]` *Number of nodes provisioned by the cloud provider*
+- `NodesReady [int]` *Number of nodes ready*
+- `Conditions [array]` *Detailed status*
+  - `Type [string]` *Type of condition - Provisioned, Ready, etc*
+  - `Status [string]` *Status of the condition - True, False, Unknown*
+  - `LastProbeTime [string]` *-optional- last time the condition was checked*
+  - `LastTransitionTime [string]` *-optional- last time the condition transitioned*
+  - `Message [string]` *-optional- Human readable message associated with the condition*
+  - `Reason [string]` *-optional- Single word camel-case reason for the condition*
+
+
+### MachineSetConfig
+  - `NodeType [NodeType:string]` *Enumerated node type for the machine set (compute or master)*
+  - `Infra [boolean]` *True if this machine set should contain infrastructure pods (registry, router)*
+  - `Size [int]` *How many machines should be created for this machine set*
+  - `Version [corev1.LocalObjectReference]` *Reference to ClusterVersion resource*
+  - `Hardware`
+    - `AWS` *-optional- contains AWS specific configuration*
+      - `InstanceType [string]` *EC2 instance type (ie. m4.xlarge)*
+      - `AMIName [string]` *Name of AMI used to create this machine set (derived from Version object)*
```

Comment:
I think it'd be good to be able to define cloud instance tags here as well (e.g. AWS instance tags).

We use instance tags for quite a few things and being able to easily define them here would be very powerful, flexible and convenient.